language-server/language-server-api/src/main/scala/aqua/lsp/LspContext.scala

```diff
@@ -118,13 +122,10 @@ object LspContext {
     ): Option[LspContext[S]] =
       // rename tokens from one context with prefix addition
       val newVariables = rename.map { renameStr =>
-        ctx.variables.map {
-          case v if v.definition.name.startsWith(name) =>
-            v.copy(definition =
-              v.definition.copy(name = v.definition.name.replaceFirst(v.definition.name, renameStr))
-            )
-
-          case kv => kv
+        ctx.variables.renameDefinitions { defName =>
+          if (defName.startsWith(name))
+            defName.replaceFirst(name, renameStr)
+          else defName
```

Comment:
nit: We can make `renameDefinitions` accept `PartialFunction` to not write `else defName`